client/src/app/components/forms/rsvp/rsvp.component.ts

```diff
@@ -52,66 +64,96 @@ export class RsvpFormComponent implements OnChanges {
 		let rsvpData: RecursivePartial<submissionData>;
 
 		//Yay!
-		if(formData.attending){
+		if (formData.attending) {
 			rsvpData = {
-				firstName : formData.firstName,
-				lastName : formData.lastName,
+				firstName: formData.firstName,
+				lastName: formData.lastName,
 				isGoing: formData.attending,
 				diet: formData.dietaryRestrictions,
-			}
+			};
 
-			if(formData.email) {
+			if (formData.email) {
 				rsvpData.email = formData.email;
 			}
 
 			//We need to parse our guests array into the correct format
 			const associates: Partial<guestData>[] = [];
 
-			for(const guest of this.guests.values()){
+			for (const guest of this.guests.values()) {
 				const associate: Partial<guestData> = {
 					firstName: guest.value.firstName,
 					lastName: guest.value.lastName,
-					diet: guest.value.dietaryRestrictions
-				}
+					diet: guest.value.dietaryRestrictions,
+				};
 
-				if(guest.value.email) {
+				if (guest.value.email) {
 					associate.email = guest.value.email;
 				}
 
 				associates.push(associate);
 			}
 
 			//Empty array isn't actually falsy
-			if(associates.length > 0) {
+			if (associates.length > 0) {
 				rsvpData.associates = associates;
 			}
-
 		}
 		//We don't need to do all that much processing if the person says no
 		else {
 			rsvpData = {
-				firstName : formData.firstName,
-				lastName : formData.lastName,
-				isGoing: formData.attending
-			}
+				firstName: formData.firstName,
+				lastName: formData.lastName,
+				isGoing: formData.attending,
+			};
 		}
 
 		this.submit.emit(rsvpData);
 	}
 
+	/**
+	 * Populate the full RSVP form with existing data about the guest and their associates
+	 */
 	populateForm() {
 		if (this.existingRsvpData) {
-			const firstName = this.existingRsvpData.firstName ?? '';
-			const lastName = this.existingRsvpData.lastName ?? '';
-			const isGoing = this.existingRsvpData.isGoing ?? false;
-			const email = this.existingRsvpData.email ?? '';
-			const diet = this.existingRsvpData.diet ?? '';
-
-			this.rsvpForm.controls.firstName.setValue(firstName);
-			this.rsvpForm.controls.lastName.setValue(lastName);
-			this.rsvpForm.controls.attending.setValue(isGoing);
-			this.rsvpForm.controls.email.setValue(email);
-			this.rsvpForm.controls.dietaryRestrictions.setValue(diet);
+			this.populateGuest(this.rsvpForm, this.existingRsvpData);
+
+			if (this.existingRsvpData.associates) {
+				for (
+					let i = 0;
+					i < this.existingRsvpData.associates.length;
+					i++
+				) {
+					if (this.guests.length < i + 1) this.addGuest();
```

Comment:
Question.

Would this be more readable as

```suggestion
					if (i >= this.guests.length) this.addGuest();
```

Reply:
Ah yeah good point